src/Orleans.Core/Logging/FileLogger.cs

```diff
@@ -27,8 +27,11 @@ public class FileLoggingOutput
         /// <param name="fileName"></param>
         public FileLoggingOutput(string fileName)
         {
-            logOutput = new StreamWriter(File.Open(fileName, FileMode.Append, FileAccess.Write, FileShare.Write));
-            this.logFileName = fileName;
+            var file = new FileInfo(fileName);
+            logFileName = file.FullName;
+            var fileExists = File.Exists(logFileName);
+            logOutput = fileExists ? file.AppendText() : file.CreateText();
```

Comment:
I would have thought we want to explicitly use UTF8 encoding,
It seems like we should use the `StreamWriter(Stream, Encoding)` ctor where the stream is `File.Open(fileName, FileMode.Append, FileAccess.Write, FileShare.ReadWrite)`. Possibly `FileShare.Read` is sufficient, but probably some apps open the file for read + write (eg, maybe if the user wants to open the file in a text editor - not sure).